src/ipv4/fragments.ml

```diff
@@ -137,11 +137,18 @@ let process cache ts (packet : Ipv4_packet.t) payload =
     match Cache.find key cache with
     | None ->
       Log.debug (fun m -> m "%a none found, inserting into cache" Ipv4_packet.pp packet) ;
-      Cache.add key (ts, packet.options, not more, 1, [(offset, payload)]) cache, None
-    | Some ((ts', options, finished, cnt, frags), cache') ->
+      let cache' =
+        Cache.add key (ts, packet.options, not more, 1, [(offset, payload)]) cache
+      in
+      Cache.trim cache', None
+    | Some (ts', options, finished, cnt, frags) ->
```

Comment:
If it was me, I'd conclude that I'm relying on adding-with-trim and add `add_trim` somewhere near the top. DRY etc.

Reply:
thanks, I revised the code in 696da0c further and discovered that e.g. `promote` wasn't needed to be called just after find..